src/main/java/com/google/cloud/tools/appengine/cloudsdk/CloudSdk.java

```diff
@@ -495,6 +487,13 @@ private void validateCloudSdkLocation() {
     }
   }
 
+  private void validateJdk() {
+    if (!Files.exists(getJavaBinaryPath())) {
```

Comment:
Good catch.

I kind of assumed it would work because, if you look at the previously existing code, it never makes Windows checks and the devappserver somehow still works...